### PR TITLE
Revert "[web] fix clicks on merged semantic nodes (#43620)"

### PR DESCRIPTION
**Pull request by yjbanov**
This reverts commit 0c1de9b8afbabe3857a3b495cc9b0bcfdf1ac305.

The commit caused https://github.com/flutter/flutter/issues/134842. I'm going to try again, this time accounting for nested clickables/tappables.

### CODE REVIEW

**Review by ditman — APPROVED** [review]
I don't think you need approvals for a revert, but here goes! RSLGTM!

**Comment by yjbanov** [comment]
The revert wasn't 100% clean, but the conflict looked small. Also, it's been some time since the commit, so I'm going to wait for the tree to go green before landing.

**Comment by ditman** [comment]
Yeah, I noticed this wasn't a 100% clean revert. Good luck with the tree!

**Pull request merged**
